lib/conjur/policy/plan.rb

```diff
@@ -2,16 +2,33 @@ module Conjur
   module Policy
     class Plan
       attr_reader :actions, :roles_created, :resources_created
-      
-      def initialize
+
+      def initialize(existing_resources, current_role_graph)
         @actions = []
         @roles_created = Set.new
         @resources_created = Set.new
+        @existing_resources = Set.new(existing_resources.collect(&:resourceid))
+        @current_role_graph = current_role_graph
+        @existing_roles = current_role_graph.inject(Set.new) {|roles, edge| roles.add(edge.parent).add(edge.child)}
       end
       
       def action a
         @actions.push a
       end
+
+      def role_exists?(id)
+        @existing_roles.include?(id)
+      end
+
+      def resource_exists?(id)
+        @existing_resources.include?(id)
+      end
+
+      def can_admin_role?(admin, role)
+        raise "no admin_option support in role graph" if @current_role_graph.first.admin_option.nil?
```

Comment:
Yes, it is. Making it backwards-compatible is what I'm working on right now.

On Tue, Jul 12, 2016 at 12:34 PM, Jon Mason <notifications@github.com>
wrote:

> In lib/conjur/policy/plan.rb
> <https://github.com/conjurinc/conjur-asset-policy/pull/19#discussion_r70473307>
> :
>
> >        end
> >
> >        def action a
> >          @actions.push a
> >        end
> > +
> > +      def role_exists?(id)
> > +        @existing_roles.include?(id)
> > +      end
> > +
> > +      def resource_exists?(id)
> > +        @existing_resources.include?(id)
> > +      end
> > +
> > +      def can_admin_role?(admin, role)
> > +        raise "no admin_option support in role graph" if @current_role_graph.first.admin_option.nil?
>
> Is admin_option nil with an older appliance version? Would it be better
> to use the api.server_version method to check this?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/conjurinc/conjur-asset-policy/pull/19/files/e43a89484ba54af6f3d0e052523f5b510cb510b6#r70473307>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEBHAz_oHmRcRGftx4zwhVF57XQZ-qSsks5qU8IigaJpZM4JGNlk>
> .
>



-- 
*Alan Potter*
Engineer
p: 617.906.8240


<http://t.yesware.com/tt/58d6ef3f2477232db08043fa9198ec0f5a0fde51/f59ded25d9f0fae3e973353a29205cf4/5ff664c66c5990e35bd8d1f3bfc0b483/www.conjur.net/>

*Identify, Authorize, Audit... Everything*
Demo
<http://t.yesware.com/tt/58d6ef3f2477232db08043fa9198ec0f5a0fde51/f59ded25d9f0fae3e973353a29205cf4/13deaa1654e874c3dccc0acce4e70193/t.sidekickopen03.com/e1t/c/5/f18dQhb0S1KL8bGch0W2n0x6l2B9nMJW7t6jwn3LqWkjN2BgL6d1vVSKf69FLZ203?t=https%3A%2F%2Finfo.conjur.net%2Fdemo&si=3680029&pi=47d6aafc-d455-4a13-9ffc-a8737ef09b32>
 | Resources
<http://t.yesware.com/tt/58d6ef3f2477232db08043fa9198ec0f5a0fde51/f59ded25d9f0fae3e973353a29205cf4/cc9a5a824c8143be8a0bab0dcc7f1c20/t.sidekickopen03.com/e1t/c/5/f18dQhb0S1KL8bGch0W2n0x6l2B9nMJW7t6jwn3LqWkjN2BgL6d1vVSKf69FLZ203?t=https%3A%2F%2Finfo.conjur.net%2Fresources&si=3680029&pi=47d6aafc-d455-4a13-9ffc-a8737ef09b32>
 | Blog
<http://t.yesware.com/tt/58d6ef3f2477232db08043fa9198ec0f5a0fde51/f59ded25d9f0fae3e973353a29205cf4/c59449660ac6f9ab10a1787fd5988490/t.sidekickopen03.com/e1t/c/5/f18dQhb0S1KL8bGch0W2n0x6l2B9nMJW7t6jwn3LqWkjN2BgL6d1vVSKf69FLZ203?t=https%3A%2F%2Fblog.conjur.net%2F&si=3680029&pi=47d6aafc-d455-4a13-9ffc-a8737ef09b32>